assets/model_monitoring/components/src/action_analyzer/action_analyzer_identify_problem_traffic/run.py

```diff
@@ -315,6 +319,7 @@ def run():
                                              args.model_deployment_name)
 
             topic_group_dict = {f"{metrics}_bad_group_{i}_{k}": (k, v) for i, (k, v) in enumerate(topics_dict.items())}
+            # [fep] bad_samples is only a subset of the bad queries, so it is possible that some bad queries do not get a topic group?
```

Comment:
I need to refactor the logic here regarding to the sample logic. If we want to sample for bad queries or not